src/linear_algebra/basic.lean

```diff
@@ -1835,6 +1835,24 @@ def of_submodules (p : submodule R M) (q : submodule R M₂) (h : p.map ↑e = q
 @[simp] lemma of_submodules_symm_apply {p : submodule R M} {q : submodule R M₂}
   (h : p.map ↑e = q) (x : q) : ↑((e.of_submodules p q h).symm x) = e.symm x := rfl
 
+/-- Given `f : M₁ ≃ₗ[R] M₂` and `U` a submodule of `M₂`, `f.comap U` is the
+induced `linear_equiv` from `U.comap f.to_linear_map` to `U`. -/
+def comap [semimodule R M] [semimodule R M₂] (f : M ≃ₗ[R] M₂) (U : submodule R M₂) :
```

Comment:
I'm thinking `restrict_comap` might be a better name, given this lemma (which I think you should add):
```lean
lemma linear_equiv.comap_to_linear_map [semimodule R M] [semimodule R M₂] (f : M ≃ₗ[R] M₂) (U : submodule R M₂) :
  (f.comap U).to_linear_map = (f.to_linear_map.dom_restrict _).cod_restrict _ subtype.prop :=
by { ext, refl }
```

Reply:
Or, alternatively. perhaps just `of_submodule'`, since really just is just `of_submodule` expressed with `comap` instead of `map`.

Reply:
I think I prefer `comap_restrict` since `of_submodule'` is not very descriptive

Reply:
Using that argument, we should rename `of_submodule` to `map_restrict`

Reply:
Okay, for consistency, lets use `of_submodule'`  then